gammapy/modeling/sampling.py

```diff
@@ -82,15 +82,24 @@ def run_mcmc(dataset, nwalkers=8, nrun=1000, threads=1):
     import emcee
 
     dataset.models.parameters.autoscale()  # Autoscale parameters
-    pars = [par.factor for par in dataset.models.parameters.free_parameters]
-    ndim = len(pars)
 
     # Initialize walkers in a ball of relative size 0.5% in all dimensions if the
     # parameters have been fit, or to 10% otherwise
+    # Handle source position spread differently with a spread of 0.1°
     # TODO: the spread of 0.5% below is valid if a pre-fit of the model has been obtained.
     # currently the run_mcmc() doesn't know the status of previous fit.
-    spread = 0.5 / 100
-    p0var = np.array([spread * pp for pp in pars])
+    p0var = []
+    pars = []
+    spread = 0.5/100
+    spread_pos = 0.1 # in degrees
```

Comment:
Is the 0.1° valid in all situations? 
Is there a possibility to guess a plausible range from the input `dataset`?

Reply:
This would be better indeed but not sure how.
The uncertainty of the source position (hence initial spread) depends on pt src/ext src, its brightness, etc
Even if 0.1° is not optimal, the walkers distribution "adapts" to the shape of the local likelihood after some run.
What is better to avoid is to have a spread too large where the init position is outside of the box or possibly taking the position of another source.